lighthouse-core/test/gather/mock-driver.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/* eslint-env jest */
+
+/**
+ * Creates a jest mock function whose implementation consumes mocked protocol responses matching the
```

Comment:
hm this file is kind of a mix of `mock-connection` and `mock-driver`, maybe we can add a file overview and/or indicate in the filename how to use what?

could be "refer to driver-test.js and blah-test.js for intended usage"

Reply:
> hm this file is kind of a mix of mock-connection and mock-driver, maybe we can add a file overview and/or indicate in the filename how to use what?

yeah, we really can't have a `fake-driver.js` and a `mock-driver.js` in the same directory :) (at least not without some heavy guidance and preferably a new name)

At a literal level, this is really just mocking connection commands and general event emitting. Maybe we could name around that and think about wrapping up in a `mock-connection.js` or something in the future?

Reply:
`mock-commands.js`